frictionless/portals/github/manager.py

```diff
@@ -1,45 +1,156 @@
-                package.add_resource(Resource(path=fullpath))
-        return package
+        paths = _get_contents(contents, repository)
+        package = _get_package(paths, repository, base_path, self.control.formats)
+        if package.resources:
+            return package
+        note = "Package not found"
+        raise FrictionlessException(note)
 
     # Write
-    # TODO: implement
+
+    def write_package(
+        self,
+        *,
+        email: Optional[str] = None,
+        repo: Optional[str] = None,
+        apikey: Optional[str] = None,
+        filename: Optional[str] = None,
+        content: Optional[str] = None,
+        branch: Optional[str] = None,
+    ):
+        repo = repo or self.control.repo
+        apikey = apikey or os.environ.get("GITHUB_ACCESS_TOKEN", None)
+        assert repo
+        assert apikey
+        # Create repo
+        repository = None
+        user = None
+        try:
+            client = platform.github.Github(apikey)
+            user = client.get_user()
+            repository = user.create_repo(
+                name=repo, auto_init=True, gitignore_template="Python"
+            )
+        except Exception as exception:
+            note = "Github API error:" + repr(exception)
+            raise FrictionlessException(note)
+
+        # Write package file to the repo
+        file_created = None
+        if content:
+            newfile = filename or "package.json"
+            repository = user.get_repo(repo)
+            email = (
+                user.email or self.control.email or os.environ.get("GITHUB_EMAIL", None)
+            )
+            username = (
+                user.name or self.control.user or os.environ.get("GITHUB_NAME", None)
+            )
+            assert email
+            assert username
+            try:
+                author = platform.github.InputGitAuthor(username, email)
+                branch = branch or repository.default_branch
+                file_created = repository.create_file(
+                    path=newfile,
+                    message="Create package.json",
+                    content=content,
+                    branch=branch,
+                    committer=author,
+                    author=author,
+                )
+            except Exception as exception:
+                note = "Github API error:" + repr(exception)
+                raise FrictionlessException(note)
+
+        response = namedtuple("response", ["repository", "file_created"])
+        return response(repository, file_created)
+
+
+def _get_contents(contents, repository):
+    paths = []
+    while contents:
+        file_content = contents.pop(0)  # type: ignore
+        if file_content.type == "dir":
+            contents.extend(repository.get_contents(file_content.path))  # type: ignore
+        else:
+            paths.append(file_content.path)
+    return paths
+
+
+def _get_package(paths, repository, base_path, formats):
```

Comment:
It's OK not to use the leading underscore here as it's an internal function anyway. We don't use the leading underscore for cases like this across the codebase